src/vma/sock/sockinfo_udp.cpp

```diff
@@ -2760,6 +2761,42 @@ size_t sockinfo_udp::handle_msg_trunc(size_t total_rx, size_t payload_size, int
 	return total_rx;
 }
 
+int sockinfo_udp::get_socket_tx_ring_fd(struct sockaddr *to, socklen_t tolen)
+{
+	NOT_IN_USE(tolen);
+	si_udp_logfunc("get_socket_tx_ring_fd fd %d to %p tolen %d", m_fd, to ,tolen);
+
+	if (!to) {
+		si_udp_logdbg("got invalid to addr null for fd %d", m_fd);
+		errno = EINVAL;
+		return -1;
+	}
+	sock_addr dst(to);
+	ring *ring = NULL;
+
+	if (m_p_connected_dst_entry && m_connected == dst) {
+		ring = m_p_connected_dst_entry->get_ring();
+	} else {
+		dst_entry_map_t::iterator it = m_dst_entry_map.begin();
+		for (; it != m_dst_entry_map.end(); it++) {
+			if (it->first == dst) {
+				ring = it->second->get_ring();
+				break;
+			}
+		}
+	}
+	if (!ring) {
+		si_udp_logdbg("could not find TX ring for fd %d addr %s",
+				m_fd, dst.to_str());
+		errno = ENODATA;
```

Comment:
Maybe ENOTCONN ?

Reply:
not good since there can be a connection but it's not offloaded